js/index.js

```diff
@@ -0,0 +1,56 @@
+"use strict";
```

Comment:
I think you've treated this class as if it's a container for your application logic - I think you should think of it as the class which knows where the bits of your application logic are. So rather than trying to control all of the view and interactions, you're just initialising the different components of your application. That will help you separate concerns I think.